### PR TITLE
Document optional dependencies

### PR DESCRIPTION
**Pull request by pmk21**
Addresses #5261 

Still working on it.
Took inspiration from pandas docs. Any suggestion about the columns to include?

@quasiben Need some help here.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Thanks for working on this. Just the package name and the min version seems fine. Off the top of my head, I would add

* gcsfs
* s3fs
* fastparquet
* pyarrow
* sqlalchemy

and maybe look for imports in the tests files to find more. There are probably others.

**Comment by pmk21** [comment]
Finding some difficulty in finding out the min version for the packages. Any easy way to do it?
And also could the following packages be added -
* scipy
* scikit-image
* cytoolz

**Comment by TomAugspurger** [comment]
I'd say whatever our lower version in our test suite is what we should
"recommend" here. There may be ones we don't pin a minimum version for.

**Comment by TomAugspurger** [comment]
Feel free to also leave the minimum version blank for any of those.

I haven't seen that appveyor failure before. Can you try merging master before pushing again?

File: docs/source/install.rst [hunk 1]
```diff
@@ -76,6 +76,38 @@ Anaconda
 
 Dask is included by default in the `Anaconda distribution <https://www.anaconda.com/download>`_.
 
+Optional dependencies
+---------------------
+
```
Comment (TomAugspurger) [inline docs/source/install.rst]: A brief introduction here would be welcome:

```suggestion

Specific functionality in Dask may require additional optional dependencies. For example, reading from Amazon S3 requires ``s3fs``. These optional dependencies and their minimum supported version are listed below.
```

**Comment by dhirschfeld** [comment]
I'd definitely mention the optional hashing algorithms:
> *It is worth pointing out that for arrays, the default hash uses python's standard sha1, but if you happen to have cityhash, xxhash or murmurhash installed (all available with pip), then you can get a large-factor speedup in the tokenisation step.*

https://github.com/dask/dask/issues/3946#issuecomment-418577510

**Comment by pmk21** [comment]
Sorry for the late update. I have added a brief introduction @TomAugspurger.

> I'd definitely mention the optional hashing algorithms:
> 
> > _It is worth pointing out that for arrays, the default hash uses python's standard sha1, but if you happen to have cityhash, xxhash or murmurhash installed (all available with pip), then you can get a large-factor speedup in the tokenisation step._
> 
> [#3946 (comment)](https://github.com/dask/dask/issues/3946#issuecomment-418577510)

@dhirschfeld Could you suggest how to include this? Maybe after the table of dependencies?

**Comment by TomAugspurger** [comment]
Those are all optional dependencies. They can go in the same table.

**Comment by pmk21** [comment]
Should I mention this after the table?

>  _It is worth pointing out that for arrays, the default hash uses python's standard sha1, but if you happen to have cityhash, xxhash or murmurhash installed (all available with pip), then you can get a large-factor speedup in the tokenisation step._

**Comment by TomAugspurger** [comment]
A column next to the minimum version with a brief description of what each
optional dependency is used for may be useful. These could say "Faster
hashing of arrays."

**Comment by pmk21** [comment]
Don't know what exactly to add for numpy and pandas descriptions. Slight help @TomAugspurger

**Comment by TomAugspurger** [comment]
Something like "required for dask.array" would be fine for those I think.

**Comment by dhirschfeld** [comment]
Also `psutil` - e.g.
https://github.com/dask/dask/blob/7f5f7fe49298c2353977b99a081be47dc28bc358/dask/dataframe/io/csv.py#L309
https://github.com/dask/dask/pull/5499#issuecomment-543242445

In particular, the Python `os.cpu_count` is broken for me so it's important for me to use `psutil` to get the correct cpu count.

Possible description:
> *Enables a more accurate CPU count, in particular in the context of a container or when cgroup limits are in place*

(depending on the resolution of that PR)

**Comment by TomAugspurger** [comment]
Sorry for the delay @pmk21, we don't get notifications when you force push.

This looks good. We can add additional libraries as we go.

**Pull request merged**